test/test.py

```diff
@@ -377,31 +380,66 @@ def test_yml_password_format_one(self, file, sparse_mirrors_with_names, checksum
         for fn in sparse_mirrors_with_names["names"]:
             assert calculate_sha256(file["tmp_path"] / fn) == checksums[str(fn)]
 
-    @pytest.mark.skipif(os.environ.get("AWS_ACCESS_KEY") in {None, ""} or os.environ.get("NO_AWS_TEST"),
+    @pytest.mark.skipif(os.environ.get("AWS_ACCESS_KEY") is None
+                        or os.environ.get("AWS_ACCESS_KEY") == ""
+                        or os.environ.get("AWS_SECRET_KEY") is None
+                        or os.environ.get("AWS_SECRET_KEY") == ""
+                        or os.environ.get("AWS_DEFAULT_REGION") is None
+                        or os.environ.get("AWS_DEFAULT_REGION") == "",
                         reason="Environment variable(s) not defined")
     def test_yml_s3_mirror(self, file, checksums, powerloader_binary):
         remove_all(file)
         out = subprocess.check_output([powerloader_binary, "download",
                                        "-f", file["pw_format_three"],
-                                       "-d", file["tmp_path"]],
-                                       env=os.environ)
+                                       "-d", file["tmp_path"]])
 
         for fp in get_files(file):
             assert calculate_sha256(fp) == checksums[str(path_to_name(fp))]
 
-    """
-    def test_yml_password_format_two(self, file, sparse_mirrors_with_names, checksums, powerloader_binary,
-                                     mock_server_working, mock_server_404, mock_server_lazy,
-                                     mock_server_broken, mock_server_password):
+    def test_s3_upload(self, file, powerloader_binary):
```

Comment:
I think you need to skip these tests if there is no `AWS_...` env var. You can copy the `skipif` from the other S3 test.

Reply:
Yep, that was the problem..